r-package/grf/bindings/AnalysisToolsBindings.cpp

```diff
@@ -155,7 +155,9 @@ Rcpp::List deserialize_tree(Rcpp::List forest_object,
     node_objects.push_back(node_object);
   }
 
-  const std::vector<size_t>& drawn_samples = tree->get_drawn_samples();
+  std::vector<size_t> drawn_samples(tree->get_drawn_samples());
+  for (size_t& index : drawn_samples) index += 1; //R is 1-indexed.
```

Comment:
Two small comments:
- For clarity, it would be nice if this logic was consistent with the for loop above for `samples`. I find the approach around `samples` a little bit more clear, since it doesn't involve mutable element references.
- The standard throughout the C++ code is to use brackets with for loops, even if it's just one line.